draft-ietf-quic-transport.md

```diff
@@ -2601,12 +2601,17 @@ connection in a recoverable state, the endpoint can send a RESET_STREAM frame
 ({{frame-reset-stream}}) with an appropriate error code to terminate just the
 affected stream.
 
-RESET_STREAM MUST be instigated by the protocol using QUIC, either directly or
-through the receipt of a STOP_SENDING frame from a peer.  RESET_STREAM carries
-an application error code.  Resetting a stream without knowledge of the
-application protocol could cause the protocol to enter an unrecoverable state.
-Application protocols might require certain streams to be reliably delivered in
-order to guarantee consistent state between endpoints.
+RESET_STREAM MUST be instigated by the protocol using QUIC.  RESET_STREAM
+carries an application error code.  Only the application protocol is able to
+cause a stream to be terminated.  A local instance of the application protocol
+uses a direct API call and a remote instance uses the STOP_SENDING frame, which
+triggers an automatic RESET_STREAM.
```

Comment:
Hm. The second sentence suggests that the determination of the application error code in the RESET_STREAM is the responsibility of the application. Fair enough. But in this last sentence, it sounds like the RESET_STREAM is generated by the transport (or some such) in response to STOP_SENDING. Also fair enough. But how does the transport determine the application error code.

I think the behavior here is fine, but the text feels... wooly, as you said. I wonder if it might be easier to read if we treated the STOP_SENDING case as more of an exception, than trying to make it part of the general "only the app resets streams" behavior.  *shrug*

Reply:
The guidance at line 657 says:
> An endpoint SHOULD copy the error code from the STOP_SENDING frame to the
> RESET_STREAM frame it sends, but MAY use any application error code.  The
> endpoint that sends a STOP_SENDING frame MAY ignore the error code carried in
> any RESET_STREAM frame it receives.

HTTP allows in one instance for the server to use a different error code if the transport implementation supports that, but everything's fine if the transport just reflects the code before the local application knows about it.